app/public/js/home.js

```diff
@@ -0,0 +1,121 @@
+//basic variables that I declared outside the function blocks to use them inside any of the two functions.
+var map;
+var geocoder;
+var gmarkers = [];
+
+//defining the central point in the map
+var central = {lat:user_latitude, lng: user_longitude};
+
+var lastProcessedAddress = '';
+
+/*
+		 check if the test field is empty of not
+		 If it is empty use the placeholder value as the user_address
+		 else use the value of the text field as the user_address
+*/
+function getAddress()
+{
+	var address = $('#address').val().trim();
+	if ( address === '' )
+	{
+		address = $('#address').attr('placeholder');
+	}
+	return address;
+}
+
+function conditionalProcessAddress() {
+  if( lastProcessedAddress.trim() !== getAddress().trim()  ) {
+		lastProcessedAddress = getAddress();
+		processAddress();
+	}
+}
+
+/*
+		This function is where I initialized the map and then called the geocodeAddress function
+		perform the geocoding and display the address.
+		I had to set the zoom to 17 because that is the only point where I carousel-indicators
+		see the names of location around the users current location.
+*/
+ function initMap() {
+     map = new google.maps.Map(document.getElementById('map'), {
+          zoom: 17,
+          center: central,
+					draggable: false
+        });
+
+	/*
+Check if the address changed every 5 seconds.  if changed, processAddress is called.
+This is to prevent overuse of the google maps api quota.
+	*/
+				window.setInterval(conditionalProcessAddress, 5000);
+}
+
+// as soon as the document is loaded, initialize the map.
+$(document).ready(initMap);
+
+/*
+Processes the user-provided address by retrieving corresponding longitude, latitude and saving the
+results through an asynchronous request to the server.
+*/
+function processAddress()
+{
+	var address = getAddress();
+
+	//creating a geocoder object
+	 geocoder = new google.maps.Geocoder();
```

Comment:
Declare geocoder as a local variable in processAddress to minimize the number of global variables.